Detectors/ITSMFT/ITS/workflow/src/CalibratorSpec.cxx

```diff
@@ -0,0 +1,1083 @@
+    const char* option = recreate ? "RECREATE" : "UPDATE";
+    TFile* tf = TFile::Open(filename.c_str(), option);
+
+    // Update the ROOT file with most recent TTree
+    tf->cd();
+    this->threshold_tree->Write(0, TObject::kOverwrite);
+
+    // Close file and clean up memory
+    tf->Close();
+  }
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Perform final operations on output objects (e.g. remove temp names)
+template <class Mapping>
+void ITSCalibrator<Mapping>::finalize_output()
+{
+
+  // In the case of a full threshold scan, rename ROOT file and
+  // create metadata file for writing to EOS
+  if (*(this->scan_type) == 'T') {
+
+    // Check that expected output directory exists
+    std::string dir = this->output_dir.empty() ? o2::utils::Str::rectifyDirectory("./") : this->output_dir;
+    dir += fmt::format("{}_{}/", this->EnvironmentID, this->run_number);
+    if (!std::filesystem::exists(dir)) {
+      LOG(error) << "Cannot find expected output directory " << dir;
+      return;
+    }
+
+    // Expected ROOT output filename
+    std::string filename = std::to_string(this->run_number) + '_' +
+                           std::to_string(this->file_number);
+    std::string filename_full = dir + filename;
+    try {
+      std::rename((filename_full + ".root.part").c_str(),
+                  (filename_full + ".root").c_str());
+    } catch (std::exception const& e) {
+      LOG(error) << "Failed to rename ROOT file " << filename_full
+                 << ".root.part, reason: " << e.what();
+    }
+
+    // Create metadata file
+    o2::dataformats::FileMetaData* md_file = new o2::dataformats::FileMetaData();
+    md_file->fillFileData(filename_full);
+    md_file->run = this->run_number;
+    md_file->LHCPeriod = this->LHC_period;
+    md_file->type = "calibration";
+    md_file->priority = "high";
+    md_file->lurl = filename_full + ".root";
+    auto metaFileNameTmp = fmt::format("{}{}.tmp", this->metafile_dir, filename);
+    auto metaFileName = fmt::format("{}{}.done", this->metafile_dir, filename);
+    try {
+      std::ofstream metaFileOut(metaFileNameTmp);
+      metaFileOut << md_file->asString() << '\n';
+      metaFileOut.close();
+      std::filesystem::rename(metaFileNameTmp, metaFileName);
+    } catch (std::exception const& e) {
+      LOG(error) << "Failed to create threshold metadata file "
+                 << metaFileName << ", reason: " << e.what();
+    }
+    delete md_file;
+  } // threshold scan
+
+  this->file_number++;
+
+  return;
+
+} // finalize_output
+
+//////////////////////////////////////////////////////////////////////////////
+// Set the run_type for this run
+// Initialize the memory needed for this specific type of run
+template <class Mapping>
+void ITSCalibrator<Mapping>::set_run_type(const short int& runtype)
+{
+
+  // Save run type info for future evaluation
+  this->run_type = runtype;
+
+  if (runtype == 42) {
```

Comment:
Perhaps it is better to use some enums instead of hardcoded identifiers?

Reply:
Thanks -- we will add an enum in the header file to define all of these numbers.